painttiming.bs

```diff
@@ -112,26 +117,35 @@ Formally, we consider the user agent to have "rendered" a document when it has p
 
     NOTE: The rendering pipeline is very complex, and the timestamp should be the latest timestamp the user agent is able to note in this pipeline (best effort). Typically the time at which the frame is submitted to the OS for display is recommended for this API.
 
-A [=generated content pseudo-element=] is a <dfn>paintable pseudo-element</dfn> when all of the following apply:
-* The pseudo-element's [=used=] [=visibility=] is <code>visible</code>.
-* The pseudo-element's [=used=] [=opacity=] is greater than zero.
-* The pseudo-element generates a non-empty [=box=].
+A [=generated content pseudo-element=] is a <dfn>paintable pseudo-element</dfn> |target| when all of the following apply:
+* |target| is [=being rendered=].
+* |target|'s [=used=] [=visibility=] is <code>visible</code>.
+* |target|'s [=used=] [=opacity=] is greater than zero.
+* |target|t generates a non-empty [=box=].
+
+A [=generated content pseudo-element=] is a <dfn>contentful pseudo-element</dfn> |target| when one or more of the following apply:
+* |target| represents a [=contentful image=].
+* |target| represents [=non-empty=] text, and its [=used=] style [=may display text=].
 
 A [=CSS image=] |img| is a <dfn>contentful image</dfn> when all of the following apply:
 * |img| is [=url valued=].
 * |img| is [=available=].
 
 A {{DOMString}} is <dfn>non-empty</dfn> if it contains at least one character excluding [=document white space characters=].
 
+A style |s| <dfn>may display text</dfn> if one of the following apply:
+* The [=foreground color=] of |s| is not [=fully transparent=].
+* The [=text-shadow=] of |s| contains at least one value that is not [=fully transparent=] and its [=length=] is greater than 0.
```

Comment:
What about colors of text decoration and text emphasis?

Reply:
I deliberately excluded them - I feel that a page that shows transparent text and colored underlines/emphasis should not be considered contentful. But this is open for discussion.

Reply:
That doesn't seem right. Text emphases and decorations aren't like borders or background colors. They're part of the text.

Reply:
They are part of the text - but if they are visible and the text isn't, is it still content rather than a mere decoration? @npm1 wdyt?

Reply:
Whipped up a [quick test page](https://jsbin.com/qawogavani/edit?html,output), and I agree that while text-shadow is content, decoration and emphasis isn't.

Reply:
Maybe the problem is with the name "may display text"? It could be "may display contentful text". or "may display glyphs".